lib/cadence/workflow/decision_task_processor.rb

```diff
@@ -86,16 +105,42 @@ def fetch_full_history
         Workflow::History.new(events)
       end
 
-      def complete_task(decisions)
+      def legacy_query_task?
+        !!task.query
+      end
+
+      def parse_queries
+        # Support for deprecated query style
+        if legacy_query_task?
+          { LEGACY_QUERY_KEY => Query.new(task.query) }
+        else
```

Comment:
Let's put `return {} if task.queries.nil?` here to catch the case where `task.queries` is nil